h3ToGeo instantiates Vector2D incorrectly

```
    public Vector2D h3ToGeo(long h3) {
        double[] coords = new double[2];
        h3Api.h3ToGeo(h3, coords);
        Vector2D out = new Vector2D(
                constrainLat(toDegrees(coords[0])),
                constrainLng(toDegrees(coords[1]))
        );
        return out;
    }
```
Is incorrect. Vector2d is defined as - `Vector2d(double x, double y)`

**Lat should be y and Lng should be x**

Tested using the following groovy script 
```
import com.uber.h3core.H3Core

H3Core h3 = H3Core.newInstance()
int res = 8
double bigBenLat = 51.5007292
double bigBenLon = -0.1246254

long index = h3.geoToH3(bigBenLat, bigBenLon, res)

double reasonableTolerance = 0.5
assert h3.h3ToGeo(index).y < (bigBenLat + reasonableTolerance) && h3.h3ToGeo(index).y > (bigBenLat + reasonableTolerance)  :
        "Oh no! The index is a long way from Big Ben. The Big Ben lat is $bigBenLat and h3 index centre is ${h3.h3ToGeo(index).y}"
assert h3.h3ToGeo(index).x < (bigBenLon + reasonableTolerance) && h3.h3ToGeo(index).x > (bigBenLon + reasonableTolerance)  :
        "Oh no! The index is a long way from Big Ben. The Big Ben lat is $bigBenLon and h3 index centre is ${h3.h3ToGeo(index).x}"
```

which gets the following stack trace 

```
Caught: java.lang.AssertionError: Oh no! The index is a long way from Big Ben. The Big Ben lat is 51.5007292 and h3 index centre is -0.1260662704929132. Expression: ((h3.h3ToGeo(index).y < (bigBenLat + reasonableTolerance)) && (h3.h3ToGeo(index).y > (bigBenLat + reasonableTolerance))). Values: bigBenLat = 51.5007292, reasonableTolerance = 0.5, bigBenLat = 51.5007292, reasonableTolerance = 0.5
java.lang.AssertionError: Oh no! The index is a long way from Big Ben. The Big Ben lat is 51.5007292 and h3 index centre is -0.1260662704929132. Expression: ((h3.h3ToGeo(index).y < (bigBenLat + reasonableTolerance)) && (h3.h3ToGeo(index).y > (bigBenLat + reasonableTolerance))). Values: bigBenLat = 51.5007292, reasonableTolerance = 0.5, bigBenLat = 51.5007292, reasonableTolerance = 0.5
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,10 @@ file [H3Core.java](./src/main/java/com/uber/h3core/H3Core.java), and support
 for the Linux x64 and Darwin x64 platforms.
 
 ## [Unreleased]
+### Changed
+- Replaced Vector2D with GeoCoord.
+### Removed
+- geoJsonOrder parameter from polyfill functions.
 
 ## [3.0.2] - 2018-05-08
 ### Fixed

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,10 +7,9 @@ file [H3Core.java](./src/main/java/com/uber/h3core/H3Core.java), and support
 for the Linux x64 and Darwin x64 platforms.
 
 ## [Unreleased]
-### Changed
+### Breaking Changes
 - Replaced Vector2D with GeoCoord.
-### Removed
-- geoJsonOrder parameter from polyfill functions.
+- Removed the geoJsonOrder parameter from polyfill functions.
 
 ## [3.0.2] - 2018-05-08
 ### Fixed

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,7 +6,7 @@ The public API of this library consists of the public functions declared in
 file [H3Core.java](./src/main/java/com/uber/h3core/H3Core.java), and support
 for the Linux x64 and Darwin x64 platforms.
 
-## [Unreleased]
+## [3.0.3] - 2018-05-22
 ### Breaking Changes
 - Replaced Vector2D with GeoCoord.
 - Removed the geoJsonOrder parameter from polyfill functions.

--- a/src/main/c/h3-java/src/jniapi.c
+++ b/src/main/c/h3-java/src/jniapi.c
@@ -381,7 +381,7 @@ JNIEXPORT void JNICALL Java_com_uber_h3core_NativeMethods_polyfill(
 
 /**
  * Converts the given polygon to managed objects
- * (ArrayList<ArrayList<ArrayList<Vector2D>>>)
+ * (ArrayList<ArrayList<ArrayList<GeoCoord>>>)
  *
  * May return early if allocation or finding required classes or methods fails.
  */
@@ -393,9 +393,9 @@ void ConvertLinkedGeoPolygonToManaged(JNIEnv *env,
         ThrowOutOfMemoryError(env);
         return;
     }
-    jclass vector2DClass =
-        (**env).FindClass(env, "com/uber/h3core/util/Vector2D");
-    if (vector2DClass == NULL) {
+    jclass geoCoordClass =
+        (**env).FindClass(env, "com/uber/h3core/util/GeoCoord");
+    if (geoCoordClass == NULL) {
         ThrowOutOfMemoryError(env);
         return;
     }
@@ -411,9 +411,9 @@ void ConvertLinkedGeoPolygonToManaged(JNIEnv *env,
         ThrowOutOfMemoryError(env);
         return;
     }
-    jmethodID vector2DConstructor =
-        (**env).GetMethodID(env, vector2DClass, "<init>", "(DD)V");
-    if (vector2DConstructor == NULL) {
+    jmethodID geoCoordConstructor =
+        (**env).GetMethodID(env, geoCoordClass, "<init>", "(DD)V");
+    if (geoCoordConstructor == NULL) {
         ThrowOutOfMemoryError(env);
         return;
     }
@@ -440,7 +440,7 @@ void ConvertLinkedGeoPolygonToManaged(JNIEnv *env,
                 LinkedGeoCoord *coord = currentLoop->first;
                 while (coord != NULL) {
                     jobject v = (**env).NewObject(
-                        env, vector2DClass, vector2DConstructor,
+                        env, geoCoordClass, geoCoordConstructor,
                         coord->vertex.lat, coord->vertex.lon);
                     if (v == NULL) {
                         return;

--- a/src/main/java/com/uber/h3core/H3Core.java
+++ b/src/main/java/com/uber/h3core/H3Core.java
@@ -16,7 +16,7 @@
 package com.uber.h3core;
 
 import com.uber.h3core.exceptions.PentagonEncounteredException;
-import com.uber.h3core.util.Vector2D;
+import com.uber.h3core.util.GeoCoord;
 
 import java.io.IOException;
 import java.util.ArrayList;
@@ -179,10 +179,10 @@ public class H3Core {
     /**
      * Find the latitude, longitude (both in degrees) center point of the cell.
      */
-    public Vector2D h3ToGeo(long h3) {
+    public GeoCoord h3ToGeo(long h3) {
         double[] coords = new double[2];
         h3Api.h3ToGeo(h3, coords);
-        Vector2D out = new Vector2D(
+        GeoCoord out = new GeoCoord(
                 constrainLat(toDegrees(coords[0])),
                 constrainLng(toDegrees(coords[1]))
         );
@@ -192,19 +192,19 @@ public class H3Core {
     /**
      * Find the latitude, longitude (degrees) center point of the cell.
      */
-    public Vector2D h3ToGeo(String h3Address) {
+    public GeoCoord h3ToGeo(String h3Address) {
         return h3ToGeo(stringToH3(h3Address));
     }
 
     /**
      * Find the cell boundary in latitude, longitude (degrees) coordinates for the cell
      */
-    public List<Vector2D> h3ToGeoBoundary(long h3) {
+    public List<GeoCoord> h3ToGeoBoundary(long h3) {
         double[] verts = new double[MAX_CELL_BNDRY_VERTS * 2];
         int numVerts = h3Api.h3ToGeoBoundary(h3, verts);
-        List<Vector2D> out = new ArrayList<>(numVerts);
+        List<GeoCoord> out = new ArrayList<>(numVerts);
         for (int i = 0; i < numVerts; i++) {
-            Vector2D coord = new Vector2D(
+            GeoCoord coord = new GeoCoord(
                     constrainLat(toDegrees(verts[i * 2])),
                     constrainLng(toDegrees(verts[(i * 2) + 1]))
             );
@@ -216,7 +216,7 @@ public class H3Core {
     /**
      * Find the cell boundary in latitude, longitude (degrees) coordinates for the cell
      */
-    public List<Vector2D> h3ToGeoBoundary(String h3Address) {
+    public List<GeoCoord> h3ToGeoBoundary(String h3Address) {
         return h3ToGeoBoundary(stringToH3(h3Address));
     }
 
@@ -414,8 +414,8 @@ public class H3Core {
      * @param holes Geofences of any internal holes
      * @param res Resolution of the desired indexes
      */
-    public List<String> polyfillAddress(List<Vector2D> points, List<List<Vector2D>> holes, int res) {
-        return polyfillAddress(points, holes, res, false);
+    public List<String> polyfillAddress(List<GeoCoord> points, List<List<GeoCoord>> holes, int res) {
+        return h3ToStringList(polyfill(points, holes, res));
     }
 
     /**
@@ -424,38 +424,14 @@ public class H3Core {
      * @param points Outline geofence
      * @param holes Geofences of any internal holes
      * @param res Resolution of the desired indexes
-     */
-    public List<Long> polyfill(List<Vector2D> points, List<List<Vector2D>> holes, int res) {
-        return polyfill(points, holes, res, false);
-    }
-
-    /**
-     * Finds indexes within the given geofence.
-     *
-     * @param points Outline geofence
-     * @param holes Geofences of any internal holes
-     * @param res Resolution of the desired indexes
-     * @param geoJsonOrder If true, accepts coordinates in (lon, lat) order as in GeoJSON.
-     */
-    public List<String> polyfillAddress(List<Vector2D> points, List<List<Vector2D>> holes, int res, boolean geoJsonOrder) {
-        return h3ToStringList(polyfill(points, holes, res, geoJsonOrder));
-    }
-
-    /**
-     * Finds indexes within the given geofence.
-     *
-     * @param points Outline geofence
-     * @param holes Geofences of any internal holes
-     * @param res Resolution of the desired indexes
-     * @param geoJsonOrder If true, accepts coordinates in (lon, lat) order as in GeoJSON.
      * @throws IllegalArgumentException Invalid resolution
      */
-    public List<Long> polyfill(List<Vector2D> points, List<List<Vector2D>> holes, int res, boolean geoJsonOrder) {
+    public List<Long> polyfill(List<GeoCoord> points, List<List<GeoCoord>> holes, int res) {
         checkResolution(res);
 
         // pack the data for use by the polyfill JNI call
         double[] verts = new double[points.size() * 2];
-        packGeofenceVertices(verts, points, 0, geoJsonOrder);
+        packGeofenceVertices(verts, points, 0);
         int[] holeSizes = new int[0];
         double[] holeVerts = new double[0];
         if (holes != null) {
@@ -469,7 +445,7 @@ public class H3Core {
             holeVerts = new double[totalSize];
             int offset = 0;
             for (int i = 0; i < holes.size(); i++) {
-                offset = packGeofenceVertices(holeVerts, holes.get(i), offset, geoJsonOrder);
+                offset = packGeofenceVertices(holeVerts, holes.get(i), offset);
             }
         }
 
@@ -487,22 +463,14 @@ public class H3Core {
      *
      * @return Next offset to begin filling from
      */
-    private static int packGeofenceVertices(double[] arr, List<Vector2D> original, int offset, boolean geoJsonOrder) {
+    private static int packGeofenceVertices(double[] arr, List<GeoCoord> original, int offset) {
         assert arr.length >= (original.size() * 2) + offset;
 
         for (int i = 0; i < original.size(); i++) {
-            double first, second;
+            GeoCoord coord = original.get(i);
 
-            if (geoJsonOrder) {
-                first = original.get(i).y;
-                second = original.get(i).x;
-            } else {
-                first = original.get(i).x;
-                second = original.get(i).y;
-            }
-
-            arr[(i * 2) + offset] = toRadians(constrainLat(first));
-            arr[(i * 2) + 1 + offset] = toRadians(constrainLng(second));
+            arr[(i * 2) + offset] = toRadians(constrainLat(coord.lat));
+            arr[(i * 2) + 1 + offset] = toRadians(constrainLng(coord.lng));
         }
 
         return (original.size() * 2) + offset;
@@ -511,7 +479,7 @@ public class H3Core {
     /**
      * Create polygons from a set of contiguous indexes
      */
-    public List<List<List<Vector2D>>> h3AddressSetToMultiPolygon(Collection<String> h3Addresses, boolean geoJson) {
+    public List<List<List<GeoCoord>>> h3AddressSetToMultiPolygon(Collection<String> h3Addresses, boolean geoJson) {
         List<Long> indices = stringToH3List(h3Addresses);
 
         return h3SetToMultiPolygon(indices, geoJson);
@@ -520,32 +488,27 @@ public class H3Core {
     /**
      * Create polygons from a set of contiguous indexes
      */
-    public List<List<List<Vector2D>>> h3SetToMultiPolygon(Collection<Long> h3, boolean geoJson) {
+    public List<List<List<GeoCoord>>> h3SetToMultiPolygon(Collection<Long> h3, boolean geoJson) {
         long[] h3AsArray = collectionToLongArray(h3);
 
-        ArrayList<List<List<Vector2D>>> result = new ArrayList<>();
+        ArrayList<List<List<GeoCoord>>> result = new ArrayList<>();
 
         h3Api.h3SetToLinkedGeo(h3AsArray, result);
 
         // For each polygon
-        for (List<List<Vector2D>> loops : result) {
+        for (List<List<GeoCoord>> loops : result) {
             // For each loop within the polygon (first being the outline,
             // further loops being "holes" or exclusions in the polygon.)
-            for (List<Vector2D> loop : loops) {
+            for (List<GeoCoord> loop : loops) {
                 // For each coordinate in the loop, we need to convert to degrees,
                 // constrain it, and ensure the correct ordering (whether geoJson
                 // or not.)
                 for (int vectorInLoop = 0; vectorInLoop < loop.size(); vectorInLoop++) {
-                    Vector2D v = loop.get(vectorInLoop);
-                    double origX = constrainLat(toDegrees(v.x));
-                    double origY = constrainLng(toDegrees(v.y));
+                    final GeoCoord v = loop.get(vectorInLoop);
+                    final double origLat = constrainLat(toDegrees(v.lat));
+                    final double origLng = constrainLng(toDegrees(v.lng));
 
-                    Vector2D replacement;
-                    if (geoJson) {
-                        replacement = new Vector2D(origX, origY);
-                    } else {
-                        replacement = new Vector2D(origY, origX);
-                    }
+                    final GeoCoord replacement = new GeoCoord(origLat, origLng);
 
                     loop.set(vectorInLoop, replacement);
                 }
@@ -898,12 +861,12 @@ public class H3Core {
     /**
      * Returns a list of coordinates representing the given edge.
      */
-    public List<Vector2D> getH3UnidirectionalEdgeBoundary(long h3) {
+    public List<GeoCoord> getH3UnidirectionalEdgeBoundary(long h3) {
         double[] verts = new double[MAX_CELL_BNDRY_VERTS * 2];
         int numVerts = h3Api.getH3UnidirectionalEdgeBoundary(h3, verts);
-        List<Vector2D> out = new ArrayList<>(numVerts);
+        List<GeoCoord> out = new ArrayList<>(numVerts);
         for (int i = 0; i < numVerts; i++) {
-            Vector2D coord = new Vector2D(
+            GeoCoord coord = new GeoCoord(
                     constrainLat(toDegrees(verts[i * 2])),
                     constrainLng(toDegrees(verts[(i * 2) + 1]))
             );
@@ -915,7 +878,7 @@ public class H3Core {
     /**
      * Returns a list of coordinates representing the given edge.
      */
-    public List<Vector2D> getH3UnidirectionalEdgeBoundary(String h3) {
+    public List<GeoCoord> getH3UnidirectionalEdgeBoundary(String h3) {
         return getH3UnidirectionalEdgeBoundary(stringToH3(h3));
     }
 

--- a/src/main/java/com/uber/h3core/NativeMethods.java
+++ b/src/main/java/com/uber/h3core/NativeMethods.java
@@ -15,7 +15,7 @@
  */
 package com.uber.h3core;
 
-import com.uber.h3core.util.Vector2D;
+import com.uber.h3core.util.GeoCoord;
 
 import java.util.ArrayList;
 import java.util.List;
@@ -48,7 +48,7 @@ final class NativeMethods {
     native int maxPolyfillSize(double[] verts, int[] holeSizes, double[] holeVerts, int res);
     native void polyfill(double[] verts, int[] holeSizes, double[] holeVerts, int res, long[] results);
 
-    native void h3SetToLinkedGeo(long[] h3, ArrayList<List<List<Vector2D>>> results);
+    native void h3SetToLinkedGeo(long[] h3, ArrayList<List<List<GeoCoord>>> results);
 
     native int compact(long[] h3, long[] results);
     native int maxUncompactSize(long[] h3, int res);

--- a/src/main/java/com/uber/h3core/util/GeoCoord.java
+++ b/src/main/java/com/uber/h3core/util/GeoCoord.java
@@ -18,33 +18,33 @@ package com.uber.h3core.util;
 import java.util.Objects;
 
 /**
- * Immutable two-dimensional vector.
+ * Immutable two-dimensional spherical coordinates, in degrees.
  */
-public class Vector2D {
-    public final double x;
-    public final double y;
+public class GeoCoord {
+    public final double lat;
+    public final double lng;
 
-    public Vector2D(double x, double y) {
-        this.x = x;
-        this.y = y;
+    public GeoCoord(double lat, double lng) {
+        this.lat = lat;
+        this.lng = lng;
     }
 
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
-        Vector2D vector = (Vector2D) o;
-        return Double.compare(vector.x, x) == 0 &&
-                Double.compare(vector.y, y) == 0;
+        GeoCoord coord = (GeoCoord) o;
+        return Double.compare(coord.lat, lat) == 0 &&
+                Double.compare(coord.lng, lng) == 0;
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(x, y);
+        return Objects.hash(lat, lng);
     }
 
     @Override
     public String toString() {
-        return String.format("Vector2D{x=%f, y=%f}", x, y);
+        return String.format("GeoCoord{lat=%f, lng=%f}", lat, lng);
     }
 }

--- a/src/test/java/com/uber/h3core/TestH3Core.java
+++ b/src/test/java/com/uber/h3core/TestH3Core.java
@@ -18,7 +18,7 @@ package com.uber.h3core;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableSet;
 import com.uber.h3core.exceptions.PentagonEncounteredException;
-import com.uber.h3core.util.Vector2D;
+import com.uber.h3core.util.GeoCoord;
 import org.junit.BeforeClass;
 import org.junit.Test;
 
@@ -68,31 +68,31 @@ public class TestH3Core {
 
     @Test
     public void testH3ToGeo() {
-        Vector2D coords = h3.h3ToGeo(22758474429497343L | (1L << 59L));
-        assertEquals(coords.x, 67.15092686397713, EPSILON);
-        assertEquals(coords.y, 191.6091114190303 - 360.0, EPSILON);
+        GeoCoord coords = h3.h3ToGeo(22758474429497343L | (1L << 59L));
+        assertEquals(coords.lat, 67.15092686397713, EPSILON);
+        assertEquals(coords.lng, 191.6091114190303 - 360.0, EPSILON);
 
-        Vector2D coords2 = h3.h3ToGeo(Long.toHexString(22758474429497343L | (1L << 59L)));
+        GeoCoord coords2 = h3.h3ToGeo(Long.toHexString(22758474429497343L | (1L << 59L)));
         assertEquals(coords, coords2);
     }
 
     @Test
     public void testH3ToGeoBoundary() {
-        List<Vector2D> boundary = h3.h3ToGeoBoundary(22758474429497343L | (1L << 59L));
-        List<Vector2D> actualBoundary = new ArrayList<>();
-        actualBoundary.add(new Vector2D(67.224749856, 191.476993415 - 360.0));
-        actualBoundary.add(new Vector2D(67.140938355, 191.373085667 - 360.0));
-        actualBoundary.add(new Vector2D(67.067252558, 191.505086715 - 360.0));
-        actualBoundary.add(new Vector2D(67.077062918, 191.740304069 - 360.0));
-        actualBoundary.add(new Vector2D(67.160561948, 191.845198829 - 360.0));
-        actualBoundary.add(new Vector2D(67.234563187, 191.713897218 - 360.0));
+        List<GeoCoord> boundary = h3.h3ToGeoBoundary(22758474429497343L | (1L << 59L));
+        List<GeoCoord> actualBoundary = new ArrayList<>();
+        actualBoundary.add(new GeoCoord(67.224749856, 191.476993415 - 360.0));
+        actualBoundary.add(new GeoCoord(67.140938355, 191.373085667 - 360.0));
+        actualBoundary.add(new GeoCoord(67.067252558, 191.505086715 - 360.0));
+        actualBoundary.add(new GeoCoord(67.077062918, 191.740304069 - 360.0));
+        actualBoundary.add(new GeoCoord(67.160561948, 191.845198829 - 360.0));
+        actualBoundary.add(new GeoCoord(67.234563187, 191.713897218 - 360.0));
 
         for (int i = 0; i < 6; i++) {
-            assertEquals(boundary.get(i).x, actualBoundary.get(i).x, EPSILON);
-            assertEquals(boundary.get(i).y, actualBoundary.get(i).y, EPSILON);
+            assertEquals(boundary.get(i).lat, actualBoundary.get(i).lat, EPSILON);
+            assertEquals(boundary.get(i).lng, actualBoundary.get(i).lng, EPSILON);
         }
 
-        List<Vector2D> boundary2 = h3.h3ToGeoBoundary(Long.toHexString(22758474429497343L | (1L << 59L)));
+        List<GeoCoord> boundary2 = h3.h3ToGeoBoundary(Long.toHexString(22758474429497343L | (1L << 59L)));
         assertEquals(boundary, boundary2);
     }
 
@@ -285,12 +285,12 @@ public class TestH3Core {
     public void testPolyfill() {
         List<Long> hexagons = h3.polyfill(
                 ImmutableList.of(
-                        new Vector2D(37.813318999983238, -122.4089866999972145),
-                        new Vector2D(37.7866302000007224, -122.3805436999997056),
-                        new Vector2D(37.7198061999978478, -122.3544736999993603),
-                        new Vector2D(37.7076131999975672, -122.5123436999983966),
-                        new Vector2D(37.7835871999971715, -122.5247187000021967),
-                        new Vector2D(37.8151571999998453, -122.4798767000009008)
+                        new GeoCoord(37.813318999983238, -122.4089866999972145),
+                        new GeoCoord(37.7866302000007224, -122.3805436999997056),
+                        new GeoCoord(37.7198061999978478, -122.3544736999993603),
+                        new GeoCoord(37.7076131999975672, -122.5123436999983966),
+                        new GeoCoord(37.7835871999971715, -122.5247187000021967),
+                        new GeoCoord(37.8151571999998453, -122.4798767000009008)
                 ), null, 9
         );
 
@@ -300,13 +300,13 @@ public class TestH3Core {
     @Test
     public void testPolyfillAddresses() {
         List<String> hexagons = h3.polyfillAddress(
-                ImmutableList.<Vector2D>of(
-                        new Vector2D(37.813318999983238, -122.4089866999972145),
-                        new Vector2D(37.7866302000007224, -122.3805436999997056),
-                        new Vector2D(37.7198061999978478, -122.3544736999993603),
-                        new Vector2D(37.7076131999975672, -122.5123436999983966),
-                        new Vector2D(37.7835871999971715, -122.5247187000021967),
-                        new Vector2D(37.8151571999998453, -122.4798767000009008)
+                ImmutableList.<GeoCoord>of(
+                        new GeoCoord(37.813318999983238, -122.4089866999972145),
+                        new GeoCoord(37.7866302000007224, -122.3805436999997056),
+                        new GeoCoord(37.7198061999978478, -122.3544736999993603),
+                        new GeoCoord(37.7076131999975672, -122.5123436999983966),
+                        new GeoCoord(37.7835871999971715, -122.5247187000021967),
+                        new GeoCoord(37.8151571999998453, -122.4798767000009008)
                 ), null, 9
         );
 
@@ -316,19 +316,19 @@ public class TestH3Core {
     @Test
     public void testPolyfillWithHole() {
         List<Long> hexagons = h3.polyfill(
-                ImmutableList.<Vector2D>of(
-                        new Vector2D(37.813318999983238, -122.4089866999972145),
-                        new Vector2D(37.7866302000007224, -122.3805436999997056),
-                        new Vector2D(37.7198061999978478, -122.3544736999993603),
-                        new Vector2D(37.7076131999975672, -122.5123436999983966),
-                        new Vector2D(37.7835871999971715, -122.5247187000021967),
-                        new Vector2D(37.8151571999998453, -122.4798767000009008)
+                ImmutableList.<GeoCoord>of(
+                        new GeoCoord(37.813318999983238, -122.4089866999972145),
+                        new GeoCoord(37.7866302000007224, -122.3805436999997056),
+                        new GeoCoord(37.7198061999978478, -122.3544736999993603),
+                        new GeoCoord(37.7076131999975672, -122.5123436999983966),
+                        new GeoCoord(37.7835871999971715, -122.5247187000021967),
+                        new GeoCoord(37.8151571999998453, -122.4798767000009008)
                 ),
-                ImmutableList.<List<Vector2D>>of(
-                        ImmutableList.<Vector2D>of(
-                                new Vector2D(37.7869802, -122.4471197),
-                                new Vector2D(37.7664102, -122.4590777),
-                                new Vector2D(37.7710682, -122.4137097)
+                ImmutableList.<List<GeoCoord>>of(
+                        ImmutableList.<GeoCoord>of(
+                                new GeoCoord(37.7869802, -122.4471197),
+                                new GeoCoord(37.7664102, -122.4590777),
+                                new GeoCoord(37.7710682, -122.4137097)
                         )
                 ),
                 9
@@ -340,24 +340,24 @@ public class TestH3Core {
     @Test
     public void testPolyfillWithTwoHoles() {
         List<Long> hexagons = h3.polyfill(
-                ImmutableList.<Vector2D>of(
-                        new Vector2D(37.813318999983238, -122.4089866999972145),
-                        new Vector2D(37.7866302000007224, -122.3805436999997056),
-                        new Vector2D(37.7198061999978478, -122.3544736999993603),
-                        new Vector2D(37.7076131999975672, -122.5123436999983966),
-                        new Vector2D(37.7835871999971715, -122.5247187000021967),
-                        new Vector2D(37.8151571999998453, -122.4798767000009008)
+                ImmutableList.<GeoCoord>of(
+                        new GeoCoord(37.813318999983238, -122.4089866999972145),
+                        new GeoCoord(37.7866302000007224, -122.3805436999997056),
+                        new GeoCoord(37.7198061999978478, -122.3544736999993603),
+                        new GeoCoord(37.7076131999975672, -122.5123436999983966),
+                        new GeoCoord(37.7835871999971715, -122.5247187000021967),
+                        new GeoCoord(37.8151571999998453, -122.4798767000009008)
                 ),
-                ImmutableList.<List<Vector2D>>of(
-                        ImmutableList.<Vector2D>of(
-                                new Vector2D(37.7869802, -122.4471197),
-                                new Vector2D(37.7664102, -122.4590777),
-                                new Vector2D(37.7710682, -122.4137097)
+                ImmutableList.<List<GeoCoord>>of(
+                        ImmutableList.<GeoCoord>of(
+                                new GeoCoord(37.7869802, -122.4471197),
+                                new GeoCoord(37.7664102, -122.4590777),
+                                new GeoCoord(37.7710682, -122.4137097)
                         ),
-                        ImmutableList.<Vector2D>of(
-                                new Vector2D(37.747976, -122.490025),
-                                new Vector2D(37.731550, -122.503758),
-                                new Vector2D(37.725440, -122.452603)
+                        ImmutableList.<GeoCoord>of(
+                                new GeoCoord(37.747976, -122.490025),
+                                new GeoCoord(37.731550, -122.503758),
+                                new GeoCoord(37.725440, -122.452603)
                         )
                 ),
                 9
@@ -373,10 +373,10 @@ public class TestH3Core {
         inputHexagons.remove(0x85283097fffffffL);
         inputHexagons.remove(0x8528309bfffffffL);
 
-        List<List<Vector2D>> geo = h3.h3SetToMultiPolygon(inputHexagons, true).get(0);
+        List<List<GeoCoord>> geo = h3.h3SetToMultiPolygon(inputHexagons, true).get(0);
 
         // TODO: looks like a bug in H3 that this is index 1
-        List<Vector2D> outline = geo.remove(1); // geo is now holes
+        List<GeoCoord> outline = geo.remove(1); // geo is now holes
 
         List<Long> outputHexagons = h3.polyfill(outline, geo, 5);
 
@@ -392,8 +392,8 @@ public class TestH3Core {
     public void testH3SetToMultiPolygonSingle() {
         long testIndex = 0x89283082837ffffL;
 
-        List<Vector2D> actualBounds = h3.h3ToGeoBoundary(testIndex);
-        List<List<List<Vector2D>>> multiBounds = h3.h3SetToMultiPolygon(ImmutableList.of(testIndex), true);
+        List<GeoCoord> actualBounds = h3.h3ToGeoBoundary(testIndex);
+        List<List<List<GeoCoord>>> multiBounds = h3.h3SetToMultiPolygon(ImmutableList.of(testIndex), true);
 
         // This is tricky, because output in an order starting from any vertex
         // would also be correct, but that's difficult to assert and there's
@@ -406,8 +406,8 @@ public class TestH3Core {
         int[] expectedIndices = {2, 3, 4, 5, 0, 1, 2};
 
         for (int i = 0; i < actualBounds.size(); i++) {
-            assertEquals(actualBounds.get(expectedIndices[i]).x, multiBounds.get(0).get(0).get(i).x, EPSILON);
-            assertEquals(actualBounds.get(expectedIndices[i]).y, multiBounds.get(0).get(0).get(i).y, EPSILON);
+            assertEquals(actualBounds.get(expectedIndices[i]).lat, multiBounds.get(0).get(0).get(i).lat, EPSILON);
+            assertEquals(actualBounds.get(expectedIndices[i]).lng, multiBounds.get(0).get(0).get(i).lng, EPSILON);
         }
     }
 
@@ -415,8 +415,8 @@ public class TestH3Core {
     public void testH3SetToMultiPolygonSingleNonGeoJson() {
         long testIndex = 0x89283082837ffffL;
 
-        List<Vector2D> actualBounds = h3.h3ToGeoBoundary(testIndex);
-        List<List<List<Vector2D>>> multiBounds = h3.h3SetToMultiPolygon(ImmutableList.of(testIndex), false);
+        List<GeoCoord> actualBounds = h3.h3ToGeoBoundary(testIndex);
+        List<List<List<GeoCoord>>> multiBounds = h3.h3SetToMultiPolygon(ImmutableList.of(testIndex), false);
 
         // This is tricky, because output in an order starting from any vertex
         // would also be correct, but that's difficult to assert and there's
@@ -429,8 +429,8 @@ public class TestH3Core {
         int[] expectedIndices = {2, 3, 4, 5, 0, 1};
 
         for (int i = 0; i < actualBounds.size(); i++) {
-            assertEquals(actualBounds.get(expectedIndices[i]).y, multiBounds.get(0).get(0).get(i).x, EPSILON);
-            assertEquals(actualBounds.get(expectedIndices[i]).x, multiBounds.get(0).get(0).get(i).y, EPSILON);
+            assertEquals(actualBounds.get(expectedIndices[i]).lat, multiBounds.get(0).get(0).get(i).lat, EPSILON);
+            assertEquals(actualBounds.get(expectedIndices[i]).lng, multiBounds.get(0).get(0).get(i).lng, EPSILON);
         }
     }
 
@@ -439,36 +439,36 @@ public class TestH3Core {
         long testIndex = 0x89283082837ffffL;
         long testIndex2 = 0x89283082833ffffL;
 
-        List<Vector2D> actualBounds = h3.h3ToGeoBoundary(testIndex);
-        List<Vector2D> actualBounds2 = h3.h3ToGeoBoundary(testIndex2);
+        List<GeoCoord> actualBounds = h3.h3ToGeoBoundary(testIndex);
+        List<GeoCoord> actualBounds2 = h3.h3ToGeoBoundary(testIndex2);
 
         // Note this is different than the h3core-js bindings, in that it uses GeoJSON (possible bug)
-        List<List<List<Vector2D>>> multiBounds = h3.h3SetToMultiPolygon(ImmutableList.of(testIndex, testIndex2), false);
+        List<List<List<GeoCoord>>> multiBounds = h3.h3SetToMultiPolygon(ImmutableList.of(testIndex, testIndex2), false);
 
         assertEquals(1, multiBounds.size());
         assertEquals(1, multiBounds.get(0).size());
         assertEquals(10, multiBounds.get(0).get(0).size());
 
-        assertEquals(actualBounds2.get(0).y, multiBounds.get(0).get(0).get(0).x, EPSILON);
-        assertEquals(actualBounds2.get(1).y, multiBounds.get(0).get(0).get(1).x, EPSILON);
-        assertEquals(actualBounds2.get(2).y, multiBounds.get(0).get(0).get(2).x, EPSILON);
-        assertEquals(actualBounds.get(1).y, multiBounds.get(0).get(0).get(3).x, EPSILON);
-        assertEquals(actualBounds.get(2).y, multiBounds.get(0).get(0).get(4).x, EPSILON);
-        assertEquals(actualBounds.get(3).y, multiBounds.get(0).get(0).get(5).x, EPSILON);
-        assertEquals(actualBounds.get(4).y, multiBounds.get(0).get(0).get(6).x, EPSILON);
-        assertEquals(actualBounds.get(5).y, multiBounds.get(0).get(0).get(7).x, EPSILON);
-        assertEquals(actualBounds2.get(4).y, multiBounds.get(0).get(0).get(8).x, EPSILON);
-        assertEquals(actualBounds2.get(5).y, multiBounds.get(0).get(0).get(9).x, EPSILON);
-        assertEquals(actualBounds2.get(0).x, multiBounds.get(0).get(0).get(0).y, EPSILON);
-        assertEquals(actualBounds2.get(1).x, multiBounds.get(0).get(0).get(1).y, EPSILON);
-        assertEquals(actualBounds2.get(2).x, multiBounds.get(0).get(0).get(2).y, EPSILON);
-        assertEquals(actualBounds.get(1).x, multiBounds.get(0).get(0).get(3).y, EPSILON);
-        assertEquals(actualBounds.get(2).x, multiBounds.get(0).get(0).get(4).y, EPSILON);
-        assertEquals(actualBounds.get(3).x, multiBounds.get(0).get(0).get(5).y, EPSILON);
-        assertEquals(actualBounds.get(4).x, multiBounds.get(0).get(0).get(6).y, EPSILON);
-        assertEquals(actualBounds.get(5).x, multiBounds.get(0).get(0).get(7).y, EPSILON);
-        assertEquals(actualBounds2.get(4).x, multiBounds.get(0).get(0).get(8).y, EPSILON);
-        assertEquals(actualBounds2.get(5).x, multiBounds.get(0).get(0).get(9).y, EPSILON);
+        assertEquals(actualBounds2.get(0).lat, multiBounds.get(0).get(0).get(0).lat, EPSILON);
+        assertEquals(actualBounds2.get(1).lat, multiBounds.get(0).get(0).get(1).lat, EPSILON);
+        assertEquals(actualBounds2.get(2).lat, multiBounds.get(0).get(0).get(2).lat, EPSILON);
+        assertEquals(actualBounds.get(1).lat, multiBounds.get(0).get(0).get(3).lat, EPSILON);
+        assertEquals(actualBounds.get(2).lat, multiBounds.get(0).get(0).get(4).lat, EPSILON);
+        assertEquals(actualBounds.get(3).lat, multiBounds.get(0).get(0).get(5).lat, EPSILON);
+        assertEquals(actualBounds.get(4).lat, multiBounds.get(0).get(0).get(6).lat, EPSILON);
+        assertEquals(actualBounds.get(5).lat, multiBounds.get(0).get(0).get(7).lat, EPSILON);
+        assertEquals(actualBounds2.get(4).lat, multiBounds.get(0).get(0).get(8).lat, EPSILON);
+        assertEquals(actualBounds2.get(5).lat, multiBounds.get(0).get(0).get(9).lat, EPSILON);
+        assertEquals(actualBounds2.get(0).lng, multiBounds.get(0).get(0).get(0).lng, EPSILON);
+        assertEquals(actualBounds2.get(1).lng, multiBounds.get(0).get(0).get(1).lng, EPSILON);
+        assertEquals(actualBounds2.get(2).lng, multiBounds.get(0).get(0).get(2).lng, EPSILON);
+        assertEquals(actualBounds.get(1).lng, multiBounds.get(0).get(0).get(3).lng, EPSILON);
+        assertEquals(actualBounds.get(2).lng, multiBounds.get(0).get(0).get(4).lng, EPSILON);
+        assertEquals(actualBounds.get(3).lng, multiBounds.get(0).get(0).get(5).lng, EPSILON);
+        assertEquals(actualBounds.get(4).lng, multiBounds.get(0).get(0).get(6).lng, EPSILON);
+        assertEquals(actualBounds.get(5).lng, multiBounds.get(0).get(0).get(7).lng, EPSILON);
+        assertEquals(actualBounds2.get(4).lng, multiBounds.get(0).get(0).get(8).lng, EPSILON);
+        assertEquals(actualBounds2.get(5).lng, multiBounds.get(0).get(0).get(9).lng, EPSILON);
     }
 
     @Test
@@ -476,7 +476,7 @@ public class TestH3Core {
         long testIndex = 0x89283082837ffffL;
         long testIndex2 = 0x8928308280fffffL;
 
-        List<List<List<Vector2D>>> multiBounds = h3.h3SetToMultiPolygon(ImmutableList.of(testIndex, testIndex2), false);
+        List<List<List<GeoCoord>>> multiBounds = h3.h3SetToMultiPolygon(ImmutableList.of(testIndex, testIndex2), false);
 
         // TODO: Update to appropriate expectations when the algorithm correctly
         // returns two polygons
@@ -490,7 +490,7 @@ public class TestH3Core {
     @Test
     public void testH3SetToMultiPolygonHole() {
         // Six hexagons in a ring around a hole
-        List<List<List<Vector2D>>> multiBounds = h3.h3AddressSetToMultiPolygon(ImmutableList.of(
+        List<List<List<GeoCoord>>> multiBounds = h3.h3AddressSetToMultiPolygon(ImmutableList.of(
                 "892830828c7ffff", "892830828d7ffff", "8928308289bffff",
                 "89283082813ffff", "8928308288fffff", "89283082883ffff"
         ), false);
@@ -511,7 +511,7 @@ public class TestH3Core {
         }
 
         // Six hexagons in a ring around a hole
-        List<List<List<Vector2D>>> multiBounds = h3.h3AddressSetToMultiPolygon(addresses, false);
+        List<List<List<GeoCoord>>> multiBounds = h3.h3AddressSetToMultiPolygon(addresses, false);
 
         assertEquals(numHexes, multiBounds.get(0).size());
     }
@@ -842,7 +842,7 @@ public class TestH3Core {
         assertEquals(6, edges.size());
         assertTrue(edges.contains(edge));
 
-        List<Vector2D> boundary = h3.getH3UnidirectionalEdgeBoundary(edge);
+        List<GeoCoord> boundary = h3.getH3UnidirectionalEdgeBoundary(edge);
         assertEquals(2, boundary.size());
     }
 

--- a/src/test/java/com/uber/h3core/TestNativeMethods.java
+++ b/src/test/java/com/uber/h3core/TestNativeMethods.java
@@ -15,7 +15,7 @@
  */
 package com.uber.h3core;
 
-import com.uber.h3core.util.Vector2D;
+import com.uber.h3core.util.GeoCoord;
 import org.junit.Test;
 
 import java.io.IOException;
@@ -40,9 +40,9 @@ public class TestNativeMethods {
         final AtomicInteger counter = new AtomicInteger(0);
 
         try {
-            nativeMethods.h3SetToLinkedGeo(new long[]{0x8928308280fffffL}, new ArrayList<List<List<Vector2D>>>() {
+            nativeMethods.h3SetToLinkedGeo(new long[]{0x8928308280fffffL}, new ArrayList<List<List<GeoCoord>>>() {
                 @Override
-                public boolean add(List<List<Vector2D>> lists) {
+                public boolean add(List<List<GeoCoord>> lists) {
                     throw new RuntimeException("crashed#" + counter.getAndIncrement());
                 }
             });

--- a/src/test/java/com/uber/h3core/benchmarking/H3CoreBenchmark.java
+++ b/src/test/java/com/uber/h3core/benchmarking/H3CoreBenchmark.java
@@ -16,7 +16,7 @@
 package com.uber.h3core.benchmarking;
 
 import com.uber.h3core.H3Core;
-import com.uber.h3core.util.Vector2D;
+import com.uber.h3core.util.GeoCoord;
 import org.openjdk.jmh.annotations.Benchmark;
 import org.openjdk.jmh.annotations.BenchmarkMode;
 import org.openjdk.jmh.annotations.Mode;
@@ -48,13 +48,13 @@ public class H3CoreBenchmark {
 
     @Benchmark
     @BenchmarkMode(Mode.Throughput)
-    public Vector2D benchmarkGetCenterCoordinates() {
+    public GeoCoord benchmarkGetCenterCoordinates() {
         return BenchmarkState.h3.h3ToGeo("85283083fffffff");
     }
 
     @Benchmark
     @BenchmarkMode(Mode.Throughput)
-    public List<Vector2D> benchmarkGetBoundary() {
+    public List<GeoCoord> benchmarkGetBoundary() {
         return BenchmarkState.h3.h3ToGeoBoundary("85283083fffffff");
     }
 

--- a/src/test/java/com/uber/h3core/benchmarking/H3SetToMultiPolygonBenchmark.java
+++ b/src/test/java/com/uber/h3core/benchmarking/H3SetToMultiPolygonBenchmark.java
@@ -17,7 +17,7 @@ package com.uber.h3core.benchmarking;
 
 import com.google.common.collect.ImmutableList;
 import com.uber.h3core.H3Core;
-import com.uber.h3core.util.Vector2D;
+import com.uber.h3core.util.GeoCoord;
 import org.openjdk.jmh.annotations.Benchmark;
 import org.openjdk.jmh.annotations.BenchmarkMode;
 import org.openjdk.jmh.annotations.Mode;
@@ -38,13 +38,13 @@ import java.util.List;
 public class H3SetToMultiPolygonBenchmark {
     @Benchmark
     @BenchmarkMode(Mode.Throughput)
-    public List<List<List<Vector2D>>> benchmarkH3SetToMultiPolygon2() {
+    public List<List<List<GeoCoord>>> benchmarkH3SetToMultiPolygon2() {
         return BenchmarkState.h3Core.h3SetToMultiPolygon(BenchmarkState.list2, false);
     }
 
     @Benchmark
     @BenchmarkMode(Mode.Throughput)
-    public List<List<List<Vector2D>>> benchmarkH3SetToMultiPolygon20() {
+    public List<List<List<GeoCoord>>> benchmarkH3SetToMultiPolygon20() {
         return BenchmarkState.h3Core.h3SetToMultiPolygon(BenchmarkState.list20, true);
     }
 

--- a/src/test/java/com/uber/h3core/benchmarking/PolyfillBenchmark.java
+++ b/src/test/java/com/uber/h3core/benchmarking/PolyfillBenchmark.java
@@ -17,7 +17,7 @@ package com.uber.h3core.benchmarking;
 
 import com.google.common.collect.ImmutableList;
 import com.uber.h3core.H3Core;
-import com.uber.h3core.util.Vector2D;
+import com.uber.h3core.util.GeoCoord;
 import org.openjdk.jmh.annotations.Benchmark;
 import org.openjdk.jmh.annotations.BenchmarkMode;
 import org.openjdk.jmh.annotations.Mode;
@@ -40,12 +40,12 @@ public class PolyfillBenchmark {
     public List<Long> benchmarkPolyfill() {
         return BenchmarkState.h3Core.polyfill(
                 ImmutableList.of(
-                        new Vector2D(37.813318999983238, -122.4089866999972145),
-                        new Vector2D(37.7866302000007224, -122.3805436999997056),
-                        new Vector2D(37.7198061999978478, -122.3544736999993603),
-                        new Vector2D(37.7076131999975672, -122.5123436999983966),
-                        new Vector2D(37.7835871999971715, -122.5247187000021967),
-                        new Vector2D(37.8151571999998453, -122.4798767000009008)
+                        new GeoCoord(37.813318999983238, -122.4089866999972145),
+                        new GeoCoord(37.7866302000007224, -122.3805436999997056),
+                        new GeoCoord(37.7198061999978478, -122.3544736999993603),
+                        new GeoCoord(37.7076131999975672, -122.5123436999983966),
+                        new GeoCoord(37.7835871999971715, -122.5247187000021967),
+                        new GeoCoord(37.8151571999998453, -122.4798767000009008)
                 ), null, 9
         );
     }
@@ -55,18 +55,18 @@ public class PolyfillBenchmark {
     public List<Long> benchmarkPolyfillWithHole() {
         return BenchmarkState.h3Core.polyfill(
                 ImmutableList.of(
-                        new Vector2D(37.813318999983238, -122.4089866999972145),
-                        new Vector2D(37.7866302000007224, -122.3805436999997056),
-                        new Vector2D(37.7198061999978478, -122.3544736999993603),
-                        new Vector2D(37.7076131999975672, -122.5123436999983966),
-                        new Vector2D(37.7835871999971715, -122.5247187000021967),
-                        new Vector2D(37.8151571999998453, -122.4798767000009008)
+                        new GeoCoord(37.813318999983238, -122.4089866999972145),
+                        new GeoCoord(37.7866302000007224, -122.3805436999997056),
+                        new GeoCoord(37.7198061999978478, -122.3544736999993603),
+                        new GeoCoord(37.7076131999975672, -122.5123436999983966),
+                        new GeoCoord(37.7835871999971715, -122.5247187000021967),
+                        new GeoCoord(37.8151571999998453, -122.4798767000009008)
                 ),
                 ImmutableList.of(
                         ImmutableList.of(
-                                new Vector2D(37.7869802, -122.4471197),
-                                new Vector2D(37.7664102, -122.4590777),
-                                new Vector2D(37.7710682, -122.4137097)
+                                new GeoCoord(37.7869802, -122.4471197),
+                                new GeoCoord(37.7664102, -122.4590777),
+                                new GeoCoord(37.7710682, -122.4137097)
                         )
                 ),
                 9
@@ -78,23 +78,23 @@ public class PolyfillBenchmark {
     public List<Long> benchmarkPolyfillWithTwoHoles() {
         return BenchmarkState.h3Core.polyfill(
                 ImmutableList.of(
-                        new Vector2D(37.813318999983238, -122.4089866999972145),
-                        new Vector2D(37.7866302000007224, -122.3805436999997056),
-                        new Vector2D(37.7198061999978478, -122.3544736999993603),
-                        new Vector2D(37.7076131999975672, -122.5123436999983966),
-                        new Vector2D(37.7835871999971715, -122.5247187000021967),
-                        new Vector2D(37.8151571999998453, -122.4798767000009008)
+                        new GeoCoord(37.813318999983238, -122.4089866999972145),
+                        new GeoCoord(37.7866302000007224, -122.3805436999997056),
+                        new GeoCoord(37.7198061999978478, -122.3544736999993603),
+                        new GeoCoord(37.7076131999975672, -122.5123436999983966),
+                        new GeoCoord(37.7835871999971715, -122.5247187000021967),
+                        new GeoCoord(37.8151571999998453, -122.4798767000009008)
                 ),
                 ImmutableList.of(
                         ImmutableList.of(
-                                new Vector2D(37.7869802, -122.4471197),
-                                new Vector2D(37.7664102, -122.4590777),
-                                new Vector2D(37.7710682, -122.4137097)
+                                new GeoCoord(37.7869802, -122.4471197),
+                                new GeoCoord(37.7664102, -122.4590777),
+                                new GeoCoord(37.7710682, -122.4137097)
                         ),
                         ImmutableList.of(
-                                new Vector2D(37.747976, -122.490025),
-                                new Vector2D(37.731550, -122.503758),
-                                new Vector2D(37.725440, -122.452603)
+                                new GeoCoord(37.747976, -122.490025),
+                                new GeoCoord(37.731550, -122.503758),
+                                new GeoCoord(37.725440, -122.452603)
                         )
                 ),
                 9

--- a/src/test/java/com/uber/h3core/util/TestGeoCoord.java
+++ b/src/test/java/com/uber/h3core/util/TestGeoCoord.java
@@ -24,12 +24,12 @@ import static org.junit.Assert.assertTrue;
 /**
  *
  */
-public class TestVector2D {
+public class TestGeoCoord {
     @Test
     public void test() {
-        Vector2D v1 = new Vector2D(0, 1);
-        Vector2D v2 = new Vector2D(1, 0);
-        Vector2D v3 = new Vector2D(0, 1);
+        GeoCoord v1 = new GeoCoord(0, 1);
+        GeoCoord v2 = new GeoCoord(1, 0);
+        GeoCoord v3 = new GeoCoord(0, 1);
 
         assertNotEquals(null, v1);
         assertNotEquals(0, v1);
@@ -44,10 +44,10 @@ public class TestVector2D {
 
     @Test
     public void testToString() {
-        Vector2D v = new Vector2D(123.456, 456.789);
+        GeoCoord v = new GeoCoord(123.456, 456.789);
 
         String toString = v.toString();
-        assertTrue(toString.contains("x=123.456"));
-        assertTrue(toString.contains("y=456.789"));
+        assertTrue(toString.contains("lat=123.456"));
+        assertTrue(toString.contains("lng=456.789"));
     }
 }

